Containers bump to Bookworm

andreriesco opened the pull request.

Commit:
bump: Bump C++, Python, Node and Rust containers to Bookworm

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>
Related-to: TIE-716

Commit:
bump: Bump Weston container to Bookworm on docker-compose.yml files

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>
Related-to: TIE-716

Commit:
tests: Fix mainFile test paths

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>
Related-to: TIE-649

andreriesco commented:
Please review @microhobby

microhobby commented:
@andreriesco please rebase it with `dev`.
Now that we have `bookwork` and `bullseye` all new changes has to be merged on `dev` first. Only fixes that are specific for `bullseye` that will be merged directly to `bullseye`.

microhobby reviewed (changes requested):
@andreriesco please rebase it

Commit:
Open dedicated terminal to create-production-image task

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

microhobby commented on cppQML/docker-compose.yml:
```diff
@@ -82,10 +79,8 @@ services:
-      # ... for tty7
-      - "c 4:7 rmw"
+      # ... for tty
+      - "c 4:* rmw"
```
tty7 is not needed anymore?

andreriesco replied:
`4:*` includes all tty ports I believe. With just 4:0 and 4:7 it was not working, I am not sure why.

andreriesco replied:
It was missing the tty1 port, now it is working

microhobby commented on dotnetAvalonia/docker-compose.yml:
```diff
@@ -55,12 +55,9 @@ services:
+    image: torizon/weston${GPU}:3
     environment:
       - ACCEPT_FSL_EULA=1
-      - WESTON_ARGS=--continue-without-input
```
I'm pretty sure that this line is yet needed...

andreriesco replied:
With this line Weston just hangs, I doesn't wait for the application container

microhobby replied:
Did you tested without mouse, keyboard or touch screen connected?

microhobby replied:
@andreriesco commented with me internally that he tested without keyboard and touch screen and it works

microhobby commented on dotnetAvalonia/docker-compose.yml:
```diff
@@ -77,10 +74,8 @@ services:
     device_cgroup_rules:
-      # ... for tty0
-      - "c 4:0 rmw"
-      # ... for tty7
```
tty7 is not needed anymore?

andreriesco replied:
`4:*` includes all tty ports I believe. With just 4:0 and 4:7 it was not working, I am not sure why.

andreriesco replied:
It was missing the tty1 port, now it is working

microhobby reviewed (commented):
@andreriesco please take a look on the review, thanks

Commit:
Add "--no-self-contained" to dotnet build and dotnet publish commands

Add "--no-self-contained" to dotnet build and dotnet publish commands.
This parameter tells the command to not compile and include the dotnet
runtime together with the compiled source code. The runtime does not
need to be included as it is already present in our dotnet Docker images
(torizon/dotnet, torizon/dotnet-debug, torizon/aspdotnet, etc). For more
information, check the dotnet publish command documentaion:
https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-publish

Related-to: TIE-716
Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
Remove "build-makedir" unused task of Qt QML deploy pipeline

Related-to: TIE-716
Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
Standardize dotnet tasks names

Related-to: TIE-716
Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
Improve the compilation and packaging of dotnet release code

Improve the compilation and packaging of dotnet release code on the
Dockerfile. With "dotnet build" and "dotnet publish" the code gets
compiled twice, because "dotnet publish" also compiles the code. Also,
by default the code gets compiled to Debug, but we want it to Release,
so the "-c Release" is necessary.
Related-to: TIE-716
Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
Update dotnet publish-torizon-${arch} arguments

Related-to: TIE-716
Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
Standardize "x86" names to "AMD64"

Related-to: TIE-716
Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
Remove build-torizon-${arch} as publish-torizon-${arch} compiles code

Removing the build-torizon-${arch} command for remote debugging, as the
publish-torizon-${arch} command already compiles the code. Therefore,
the code was being compiled twice.

Related-to: TIE-716
Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

microhobby commented on assets/tasks/common.json:
```diff
@@ -612,32 +614,6 @@
             }
         },
-        {
-            "label": "pre-cleanup-x86",
```
I see that you remove it, maybe to change to `pre-cleanup-amd64`, but I cannot find the entry for it in the additions...

andreriesco replied:
This task was duplicated, so I just removed this one. https://github.com/andreriesco/vscode-torizon-templates/blob/tie-716/assets/tasks/common.json#L591

microhobby commented on cppConsole/.vscode/tasks.json:
```diff
@@ -450,16 +450,16 @@
-                "push-container-torizon-debug-x86",
-                "pull-container-torizon-debug-x86",
-                "run-container-torizon-debug-x86",
+                "pre-cleanup-amd64",
```
For example, here we are depending on `pre-cleanup-amd64` that does not exists

andreriesco replied:
This task was duplicated, so I just removed the duplicated one but it is still there. https://github.com/andreriesco/vscode-torizon-templates/blob/tie-716/assets/tasks/common.json#L591
